src/MongoDB.EntityFrameworkCore/Serializers/SerializationHelper.cs

```diff
@@ -44,25 +45,63 @@ public static void WriteProperties(BsonDocument document, IUpdateEntry entry, IE
         using var writer = new BsonDocumentWriter(document);
         writer.WriteStartDocument();
 
+        // Write PK first, including all primary key properties in case of composite key
+        if (properties.Any(p => p.IsPrimaryKey()))
+        {
+            var pk = entry.EntityType.FindPrimaryKey();
+            if (pk.Properties.Count > 1)
+            {
+                writer.WriteName("_id");
+                writer.WriteStartDocument();
+            }
+
+            foreach (var property in pk.Properties)
+            {
+                var propertyValue = entry.GetCurrentValue(property);
+                var serializationInfo = GetPropertySerializationInfo(property);
+                var elementName = serializationInfo.ElementPath?.Last() ?? serializationInfo.ElementName;
+                WriteProperty(writer, elementName, propertyValue, serializationInfo.Serializer);
+            }
+
+            if (pk.Properties.Count > 1)
+            {
+                writer.WriteEndDocument();
+            }
+        }
+
         foreach (var property in properties)
         {
-            var propertyValue = entry.GetCurrentValue(property);
-            var propertySerializationInfo = GetPropertySerializationInfo(property);
-            // TODO: Add ElementPath support here
-            var elementName = propertySerializationInfo.ElementName;
-            writer.WriteName(elementName);
+            if (property.IsPrimaryKey())
```

Comment:
I personally prefer putting these into the foreach loop - e.g. `properties.Where(p => !p.IsPrimaryKey())`. Not a show stopper though.